### PR TITLE
js_2048_game: add tests

### PR DESCRIPTION
**Pull request by brainwashers**

### CODE REVIEW

File: cypress/integration/js2048Game.spec.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+  describe('2048 game', () => {
+    beforeEach('Open site', () => {
+      cy.visit('/');
+    });
```
Comment (roman-mate) [inline cypress/integration/js2048Game.spec.js:6]: https://docs.cypress.io/guides/references/best-practices.html#Setting-a-global-baseUrl

File: cypress/integration/js2048Game.spec.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      cy.get('button').click();
+      cy.get('body').type('{leftArrow}');
+      cy.get('body').type('{downArrow}');
+      cy.get('body').type('{leftArrow}');
+      cy.get('[class="game-score"]').should('not.have.value', '0');
+      cy.get('button').click();
+      cy.get('[class="game-score"]').should('have.value', '');
+    });
+
+    it('should show message wh', () => {
```
Comment (roman-mate) [inline cypress/integration/js2048Game.spec.js:50]: incomplete text?

File: cypress/integration/js2048Game.spec.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+  describe('2048 game', () => {
+    beforeEach('Open site', () => {
+      cy.visit('/');
+    });
+
+    it('should show the score', () => {
+      cy.get('button').click();
```
Comment (roman-mate) [inline cypress/integration/js2048Game.spec.js:9]: Is there a better selector?

**Review by roman-mate — CHANGES_REQUESTED** [review]

**Review by rhonyshniuk — APPROVED** [review]

File: cypress/integration/js2048Game.spec.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  });
+
+  it('should show the initial score', () => {
+    cy.get('.button.start').click();
+    cy.get('.game-score').should('have.value', '');
+  });
+
+  it('should show the score', () => {
+    cy.get('.button.start').click();
+    cy.get('body').type('{leftArrow}');
```
Comment (rhonyshniuk) [inline cypress/integration/js2048Game.spec.js:15]: Add loop there or move into a function

File: cypress/integration/js2048Game.spec.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+/// <reference types="cypress" />
```
Comment (roman-mate) [inline cypress/integration/js2048Game.spec.js:1]: Base on what I read this is needed for typescript. Is this needed for js?

File: cypress/integration/js2048Game.spec.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+'use strict';
+
+Cypress.Commands.add('shuffleBoxes', (arrow1, arrow2, times) => {
+  cy.get('.button.start').click();
+
+  for (let n = 0; n < times; n++) {
+    cy.get('body').type(arrow1);
+    cy.get('body').type(arrow2);
+  }
+  cy.get('.game-score').invoke('text').then(parseFloat).should('be.gt', 1);
```
Comment (roman-mate) [inline cypress/integration/js2048Game.spec.js:11]: if you shuffle 1 time, the score is unlikely to be >1. So how about we move this check outside of shuffleBoxes method?

File: cypress/integration/js2048Game.spec.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+    for (let n = 0; n < 100; n++) {
+      cy.get('body').type('{rightArrow}');
+      cy.get('body').type('{downArrow}');
+      cy.get('body').type('{leftArrow}');
+      cy.get('body').type('{upArrow}');
+    }
+
+    cy.contains('You lose! Restart the game?')
+      .should('not.have.class', 'hidden');
```
Comment (roman-mate) [inline cypress/integration/js2048Game.spec.js:57]: can we just check it's visible?

File: cypress/integration/js2048Game.spec.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+  it('should show message in case of the loss', () => {
+    cy.get('.button.start').click();
+
+    for (let n = 0; n < 100; n++) {
+      cy.get('body').type('{rightArrow}');
+      cy.get('body').type('{downArrow}');
+      cy.get('body').type('{leftArrow}');
+      cy.get('body').type('{upArrow}');
+    }
```
Comment (roman-mate) [inline cypress/integration/js2048Game.spec.js:54]: if you move verification that score >1 from the shuffleBoxed method, then you'll be able to reuse that method here.

File: cypress/integration/js2048Game.spec.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    cy.shuffleBoxes('{rightArrow}', '{upArrow}', 3);
+  });
+
+  it.only('should show the score', () => {
+    cy.shuffleBoxes('{rightArrow}', '{downArrow}', 3);
+  });
+
+  it.only('should show the score', () => {
+    cy.shuffleBoxes('{leftArrow}', '{downArrow}', 3);
+  });
```
Comment (roman-mate) [inline cypress/integration/js2048Game.spec.js:38]: Can we have different descriptions representing what exactly we are testing in each test case?

File: cypress/integration/js2048Game.spec.js (hunk 5 above)
Comment (brainwashers) [reply cypress/integration/js2048Game.spec.js:1]: I use it for autocompletion

File: cypress/integration/js2048Game.spec.js (hunk 8 above)
Comment (brainwashers) [reply cypress/integration/js2048Game.spec.js:54]: I want to use here 4 arrows. can't do this with shuffleBoxes method

**Review by rhonyshniuk — APPROVED** [review]

File: cypress/integration/js2048Game.spec.js (hunk 6 above)
Comment (roman-mate) [reply cypress/integration/js2048Game.spec.js:10]: remove this check

File: cypress/integration/js2048Game.spec.js (hunk 8 above)
Comment (roman-mate) [reply cypress/integration/js2048Game.spec.js:58]: user shuffle 100 times.

File: cypress/integration/js2048Game.spec.js (hunk 6 above)
Comment (brainwashers) [reply cypress/integration/js2048Game.spec.js]: done

**Review by roman-mate — APPROVED** [review]

**Pull request merged**